hack/jenkins/e2e.sh

```diff
@@ -262,6 +257,8 @@ if [[ ${JOB_NAME} =~ ^kubernetes-.*-gce ]]; then
 elif [[ ${JOB_NAME} =~ ^kubernetes-.*-gke ]]; then
   KUBERNETES_PROVIDER="gke"
   : ${E2E_ZONE:="us-central1-f"}
+  : ${DOGFOOD_GCLOUD:="true"}
```

Comment:
You can actually remove this variable entirely, it isn't used anymore. Used to be for choosing the command group, i.e. alpha, beta.

To use dogfood gcloud set `${CLOUDSDK_BUCKET:="gs://cloud-sdk-build/testing/rc"}`